### DISCUSSION TITLE
How to see current version of particular grammar?

### DISCUSSION THREAD

**Discussion by madskjeldgaard** [nvim-treesitter/nvim-treesitter#1303] [Q&A]
Hello all
I was wondering if there was a feature that would showcase the installed version of a particular grammar? I am trying to debug an update situation here and TSInstallInfo only lets me know if it is installed or not. 
What I am struggling with right now is why nvim-treesitter is not updating a grammar (supercollider) that has been updated. Is there something that needs to be done in the nvim-treesitter repo before the update propagates?

Thanks!

**Comment by theHamsta** [comment]
The current revision is in `lockfile.json` and the current revision that you installed in the `parser_info` folder (currently not represented in the TUI).

We're pushing new revisions after each update of nvim-treesitter except there's something that needs to be resolved manually
https://github.com/nvim-treesitter/nvim-treesitter/pull/1300/files

I had problems with deactivating the `requires_generate_from_grammar` option of super collider. It sees that some of the C files where not fully updated because the ABI is still 12

**Comment by theHamsta** [comment]
This function gives you the currently installed revision: https://github.com/nvim-treesitter/nvim-treesitter/blob/4dcc3fb4b38d4a29d175fda69a0bf35b2aa675c5/lua/nvim-treesitter/install.lua#L42

PRs to display it in TSInstallInfo are welcome.

**Comment by theHamsta** [comment]
I just updated supercollider to the most recent revision btw.

**Comment by madskjeldgaard** [comment]
Thanks a lot for the help and the clarifications. Now it makes more sense - I understand the update workflow a bit better. I'll look into the ABI stuff. Thanks again!
